xmake.lua

```diff
@@ -32,10 +32,50 @@ add_rules("ue4ss.core")
 -- Restrict the compilation modes/configs.
 -- These restrictions are inherited upstream and downstream.
 -- Any project that `includes("UE4SS")` will inherit these global restrictions.
-set_allowedplats("windows")
-set_allowedarchs("x64")
+set_allowedplats("windows", "linux")
+set_allowedarchs("x64", "x86_64", "x86_64-unknown-linux-gnu")
 set_allowedmodes(modes)
 
+option("zig")
+    set_showmenu(true)
+    set_description("Use the Zig compiler.")
+    set_default(false)
+option_end()
+
+toolchain("zigcross")
+    if is_host("windows") then
+        set_toolset("cc", "$(projectdir)/tools/zig/zig-cc.bat")
+        set_toolset("cxx", "$(projectdir)/tools/zig/zig-c++.bat")
+        set_toolset("ld", "$(projectdir)/tools/zig/zig-c++.bat")
+        set_toolset("sh", "$(projectdir)/tools/zig/zig-c++.bat")
+        set_toolset("ar", "$(projectdir)/tools/zig/zig-ar.bat")
+    else
+        set_toolset("cc", "$(projectdir)/tools/zig/zig-cc")
+        set_toolset("cxx", "$(projectdir)/tools/zig/zig-c++")
+        set_toolset("ld", "$(projectdir)/tools/zig/zig-c++")
+        set_toolset("sh", "$(projectdir)/tools/zig/zig-c++")
+        set_toolset("ar", "$(projectdir)/tools/zig/zig-ar")
+    end
+    add_cxflags("-fexperimental-library")
+    add_cxflags("-fno-delete-null-pointer-checks")
+    add_cxflags("-gdwarf")
+    add_cxflags("-fno-sanitize=undefined") -- can also use O2 to avoid this, but I'd prefer getting clear binary for now
+    add_shflags("-z", "lazy")
+toolchain_end()
+
+if is_plat("linux") then
```

Comment:
This should be `is_host("linux")`. I have a PR open in UE4SS repo that fixes this for windows.

Reply:
no, it's meant to be `is_plat("linux")` so it also works when cross compile from windows. 
Windows user only need zig + gnu make to compile uiless linux version on windows, so it can simplify the modder who want to create cpp mod works on both platform.